src/openrct2-win/openrct2-win.cpp

```diff
@@ -17,64 +17,50 @@
     "\"/manifestdependency:type='win32' name='Microsoft.Windows.Common-Controls' version='6.0.0.0' processorArchitecture='*' publicKeyToken='6595b64144ccf1df' language='*'\"")
 
 // Then the rest
+#include <algorithm>
 #include <openrct2-ui/Ui.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string>
+#include <vector>
 
-static char** GetCommandLineArgs(int argc, wchar_t** argvW);
-static void FreeCommandLineArgs(int argc, char** argv);
-static char* ConvertWideChartoUTF8(const wchar_t* src);
+static std::vector<std::string> GetCommandLineArgs(int argc, wchar_t** argvW);
+static std::string ConvertWideChartoUTF8(const wchar_t* src);
 
 /**
  * Windows entry point to OpenRCT2 with a console window using a traditional C main function.
  */
 int wmain(int argc, wchar_t** argvW, [[maybe_unused]] wchar_t* envp)
 {
-    auto argv = GetCommandLineArgs(argc, argvW);
-    if (argv == nullptr)
-    {
-        puts("Unable to fetch command line arguments.");
-        return -1;
-    }
+    auto argvStrings = GetCommandLineArgs(argc, argvW);
 
     SetConsoleCP(CP_UTF8);
     SetConsoleOutputCP(CP_UTF8);
-    auto exitCode = NormalisedMain(argc, const_cast<const char**>(argv));
 
-    FreeCommandLineArgs(argc, argv);
+    std::vector<const char*> argv;
+    std::transform(
+        argvStrings.begin(), argvStrings.end(), std::back_inserter(argv), [](const auto& string) { return string.c_str(); });
+    auto exitCode = NormalisedMain(argc, argv.data());
     return exitCode;
 }
 
-static char** GetCommandLineArgs(int argc, wchar_t** argvW)
+static std::vector<std::string> GetCommandLineArgs(int argc, wchar_t** argvW)
 {
     // Allocate UTF-8 strings
-    auto argv = static_cast<char**>(malloc(argc * sizeof(char*)));
-    if (argv != nullptr)
-    {
-        // Convert to UTF-8
-        for (int i = 0; i < argc; i++)
-        {
-            argv[i] = ConvertWideChartoUTF8(argvW[i]);
-        }
-    }
-    return argv;
-}
-
-static void FreeCommandLineArgs(int argc, char** argv)
-{
-    // Free argv
+    std::vector<std::string> argv(argc);
     for (int i = 0; i < argc; i++)
     {
-        free(argv[i]);
+        argv[i] = ConvertWideChartoUTF8(argvW[i]);
     }
-    free(argv);
+    return argv;
 }
 
-static char* ConvertWideChartoUTF8(const wchar_t* src)
+static std::string ConvertWideChartoUTF8(const wchar_t* src)
```

Comment:
How is this different to https://github.com/OpenRCT2/OpenRCT2/blob/9ef8d6da421399e69a687bcaf3a596702ce27009/src/openrct2/core/String.cpp#L72

Can't we use that function.